kevm-pyk/src/kevm_pyk/__main__.py

```diff
@@ -296,43 +302,54 @@ def exec_foundry_prove(
     definition_dir = foundry_out / 'kompiled'
     use_directory = foundry_out / 'specs'
     use_directory.mkdir(parents=True, exist_ok=True)
-    kevm = KEVM(definition_dir, profile=profile, use_directory=use_directory)
     kcfgs_file = definition_dir / 'kcfgs.json'
-    kcfgs: Dict[str, KCFG] = {}
+
     _LOGGER.info(f'Reading file: {kcfgs_file}')
     with open(kcfgs_file, 'r') as kf:
         kcfgs = {k: KCFG.from_dict(v) for k, v in json.loads(kf.read()).items()}
+
     tests = [Contract.contract_test_to_claim_id(_t) for _t in tests]
     exclude_tests = [Contract.contract_test_to_claim_id(_t) for _t in exclude_tests]
-    claims = list(kcfgs.keys())
+    claim_names = list(kcfgs.keys())
     _unfound_kcfgs: List[str] = []
     if len(tests) > 0:
         kcfgs = {k: kcfg for k, kcfg in kcfgs.items() if k in tests}
     for _t in tests:
-        if _t not in claims:
+        if _t not in claim_names:
             _unfound_kcfgs.append(_t)
     for _t in exclude_tests:
-        if _t not in claims:
+        if _t not in claim_names:
             _unfound_kcfgs.append(_t)
         if _t in kcfgs:
             kcfgs.pop(_t)
     if _unfound_kcfgs:
         _LOGGER.error(f'Missing KCFGs for tests: {_unfound_kcfgs}')
         sys.exit(1)
-    _failed_claims: List[str] = []
-    for kcfg_name, kcfg in kcfgs.items():
-        _LOGGER.info(f'Proving KCFG: {kcfg_name}')
-        edge = kcfg.create_edge(kcfg.get_unique_init().id, kcfg.get_unique_target().id, mlTop(), depth=-1)
-        claim = edge.to_claim()
-        result = kevm.prove_claim(claim, kcfg_name.replace('.', '-'))
-        if type(result) is KApply and result.label.name == '#Top':
-            _LOGGER.info(f'Proved KCFG: {kcfg_name}')
-        else:
-            _LOGGER.error(f'Failed to prove KCFG: {kcfg_name}')
-            _failed_claims.append(kcfg_name)
-    if _failed_claims:
-        _LOGGER.error(f'Failed to prove KCFGs: {_failed_claims}')
-        sys.exit(1)
+    claims = [
+        (
+            kcfg_name.replace('.', '-'),
+            kcfg.create_edge(kcfg.get_unique_init().id, kcfg.get_unique_target().id, mlTop(), depth=-1).to_claim(),
+        )
+        for kcfg_name, kcfg in kcfgs.items()
+    ]
+
+    kevm = KEVM(definition_dir, profile=profile, use_directory=use_directory)
+
+    def prove_it(_id_and_claim: Tuple[str, KClaim]) -> Tuple[bool, KInner]:
+        _claim_id, _claim = _id_and_claim
+        return KProve_prove_claim(kevm, _claim, _claim_id, _LOGGER)
+
+    with ProcessPool(ncpus=parallel) as process_pool:
+        results = process_pool.map(prove_it, claims)
+        process_pool.close()
+
+    failed_claims = [(cid, result) for ((cid, _), (failed, result)) in zip(claims, results) if failed]
+    _failed_claim_ids = [cid for cid, _ in failed_claims]
+
+    if _failed_claim_ids:
+        _LOGGER.error(f'Failed to prove KCFGs: {_failed_claim_ids}')
```

Comment:
I think this is part of ordinary usage of the program, consider using `print` instead ([docs](https://docs.python.org/3/howto/logging.html#when-to-use-logging)).

Reply:
Done.